src/patternfly/components/Toolbar/toolbar.scss

```diff
@@ -27,67 +28,120 @@ $pf-v6--include-toolbar-breakpoints: true !default;
+  --#{$toolbar}--m-sticky--PaddingBlockEnd: var(--pf-t--global--spacer--md);
   --#{$toolbar}--m-sticky--BoxShadow: var(--pf-t--global--box-shadow--md--bottom);
   --#{$toolbar}--m-sticky--BackgroundColor: var(--pf-t--global--background--color--primary--default);
 
-  // * Toolbar insets
-  --#{$toolbar}--m-page-insets--inset: var(--#{$page}--inset);
-
   // * Toolbar expand all
   --#{$toolbar}__expand-all-icon--Rotate: 0;
   --#{$toolbar}__expand-all-icon--Transition: var(--pf-t--global--transition);
   --#{$toolbar}__item--m-expand-all--m-expanded__expand-all-icon--Rotate: 90deg;
 
   // * Toolbar filter group
-  --#{$toolbar}__group--m-filter-group--ColumnGap: var(--pf-t--global--spacer--xs);
+  --#{$toolbar}__group--m-filter-group--ColumnGap: var(--pf-t--global--spacer--gap--control-to-control--default);
 
   // * Toolbar label
   --#{$toolbar}__item--m-label--FontWeight: var(--pf-t--global--font--weight--body--bold);
+
+  // * Toolbar group label group
+  --#{$toolbar}__group--m-label-group--ColumnGap: var(--pf-t--global--spacer--gap--text-to-element--default);
+
+  // * Toolbar group action group
+  --#{$toolbar}__group--m-action-group--ColumnGap: var(--pf-t--global--spacer--gap--action-to-action--default);
+
+  // * Toolbar group action group plain
+  --#{$toolbar}__group--m-action-group-plain--ColumnGap: var(--pf-t--global--spacer--gap--action-to-action--plain);
+
+  // * Toolbar group action group inline
+  --#{$toolbar}__group--m-action-group-inline--ColumnGap: var(--pf-t--global--spacer--gap--action-to-action--default);
 }
 
-// - Toolbar content - Toolbar group - Toolbar item - Toolbar group label container - Toolbar group label group
+// - Toolbar content section - Toolbar group - Toolbar item
 .#{$toolbar}__content-section,
 .#{$toolbar}__group,
 .#{$toolbar}__item {
   @include pf-v6-hidden-visible(flex);
+}
 
-  flex-wrap: wrap;
+// - Toolbar - Toolbar content
+.#{$toolbar},
+.#{$toolbar}__content {
+  position: relative;
 }
 
-// - Toolbar content
+// - Toolbar content - Toolbar content section - Toolbar expandable content
 .#{$toolbar}__content {
   @include pf-v6-hidden-visible(grid);
+
+  row-gap: var(--#{$toolbar}__content--RowGap);
+  padding-inline-start: var(--#{$toolbar}__content--PaddingInlineStart);
+  padding-inline-end: var(--#{$toolbar}__content--PaddingInlineEnd);
 }
 
-// - Toolbar - Toolbar content
-.#{$toolbar},
-.#{$toolbar}__content {
-  position: relative;
+// Toolbar content section
+.#{$toolbar}__content-section {
+  flex-wrap: wrap;
+  row-gap: var(--#{$toolbar}__content-section--RowGap);
+  column-gap: var(--#{$toolbar}__content-section--ColumnGap);
+  align-items: start;
+
+  // push pagination to inline-end when flex items are not wrapped
+  &:has(.pf-m-pagination:last-child) {
+    .#{$toolbar}__item.pf-m-pagination {
+      margin-inline-start: revert;
+    }
+
+    > .#{$toolbar}__group,
+    > .#{$toolbar}__item {
+      &:nth-last-child(2) {
+        flex: 1;
+      }
+    }
```

Comment:
This `:nth-last-child(2)` pseudo seems like it could be fragile? Can you point to an example that shows what this is targeting?